app/src/main/java/com/example/tasky/di/RetrofitModule.kt

```diff
@@ -0,0 +1,35 @@
+package com.example.tasky.di
+
+import dagger.Module
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+class RetrofitModule {
+
+    @Singleton
+    fun providesOkHttpClient() =
+        OkHttpClient
+            .Builder()
+            .addInterceptor(
+                HttpLoggingInterceptor()
+                    .setLevel(HttpLoggingInterceptor.Level.BODY)
+            )
+            .build()
+
+
+    @Singleton
+    fun providesRetrofit(client: OkHttpClient) =
+        Retrofit
+            .Builder()
+            .baseUrl("https://tasky.pl-coding.com/")
```

Comment:
This is a personal thing, but I like to have the url either on the RetrofitService interface, or on the build.gradle

Reply:
I put it in the Retrofit interface, but as long as it's only used at a single place a string like this already serves like a constant

Reply:
Im not sure what you both mean with "Retrofit interface". Do you mean when I create a Retrofit interface with a method to cosume the API?

Reply:
Yea exactly!
That's the interface we are talking about

Example:
![image](https://github.com/MelvinKucuk/tasky/assets/16141845/9c82d47f-87bb-4390-a650-94af9da8ae28)

Reply:
And its ok to have only one Retrofit interface for all the api calls? I am used to create different Retrofit interfaces by domain. For example group the ones regarding to login or the ones regarding events. Then should I have the base URL in all of them? I prefer the approach of making a constant in the hilt module or on the build.gradle